packages/swapper/src/swappers/PortalsSwapper/getPortalsTradeQuote/getPortalsTradeQuote.ts

```diff
@@ -116,11 +116,36 @@ export async function getPortalsTradeQuote(
       partner: getTreasuryAddressFromChainId(sellAsset.chainId),
       feePercentage: affiliateBpsPercentage,
       validate: true,
-    }).catch(e => {
+    }).catch(async e => {
+      // If validation fails, fire two more quotes:
+      // 1. a quote with validation enabled, but using a well-funded address to get a rough gasLimit estimate
+      // 2. another quote with validation disabled, to get an actual quote
       console.info('failed to get Portals quote with validation enabled', e)
+      const dummyQuoteParams = getDummyQuoteParams(sellAsset.chainId)
 
-      // If validation fails, try again without validation, we won't get network fees, but we can't do any better
-      return fetchPortalsTradeOrder({
+      const dummySellAssetAddress = fromAssetId(dummyQuoteParams.sellAssetId).assetReference
+      const dummyBuyAssetAddress = fromAssetId(dummyQuoteParams.buyAssetId).assetReference
+
+      const dummyInputToken = `${portalsNetwork}:${dummySellAssetAddress}`
+      const dummyOutputToken = `${portalsNetwork}:${dummyBuyAssetAddress}`
+
+      const maybeGasLimit = await fetchPortalsTradeOrder({
+        sender: dummyQuoteParams.accountAddress,
+        inputToken: dummyInputToken,
+        outputToken: dummyOutputToken,
+        inputAmount: dummyQuoteParams.sellAmountCryptoBaseUnit,
+        slippageTolerancePercentage: 10, // hardcoded high slippage tolerance for the dummy quote, the actual one will use the correct one
```

Comment:
i.e given the pairs/low amounts, slippage may be quite high, this ensures we get a quote (we're not concerned about the actual rate here, only the `gasLimit`)

Reply:
Ok, makes sense